scene/main/viewport.cpp

```diff
@@ -1033,6 +1033,14 @@ bool Viewport::_set_size(const Size2i &p_size, const Size2i &p_size_2d_override,
 	Rect2i limit = get_visible_rect();
 	for (int i = 0; i < gui.sub_windows.size(); ++i) {
 		Window *sw = gui.sub_windows[i].window;
+#ifdef TOOLS_ENABLED
+		if (!is_part_of_edited_scene() && sw->is_part_of_edited_scene()) {
+			continue;
+		}
+#endif
```

Comment:
Should the `is_part_of_edited_scene()` call be moved out of the loop?
- pro: check is executed only once
- con: implementation with two #`ifdef`-blocks is not as nice as a single block inside the loop.

It's probably better to not implement this suggestion.

```suggestion
#ifdef TOOLS_ENABLED
	if (!is_part_of_edited_scene()) {
		return true;
	}
#endif
	Rect2i limit = get_visible_rect();
	for (int i = 0; i < gui.sub_windows.size(); ++i) {
		Window *sw = gui.sub_windows[i].window;
#ifdef TOOLS_ENABLED
		if (sw->is_part_of_edited_scene()) {
			continue;
		}
#endif
```

Reply:
That looks like it changes the logic. It would have to store `is_part_of_edited_scene()` into a var and keep the check in the loop, since it is `&&` not `||`.
But I decided to have it not use a variable for the same reason.

Reply:
You are absolutely right. In the future I should refrain from reviewing PRs on mobile. Sorry for the noise!